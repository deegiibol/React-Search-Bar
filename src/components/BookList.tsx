import React, { Component } from 'react';
import BookContainer from './BookContainer';
import { IBook } from './Models';

interface PropType {
  books: IBook[];
}

class BookList extends Component<PropType> {
  render() {
    const bookList = this.props.books.map((book) => (
      <BookContainer key={book.id} {...book} />
    ));
    return <div className="booklist">{bookList}</div>;
  }
}

export default BookList;
