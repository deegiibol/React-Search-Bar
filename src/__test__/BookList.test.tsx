import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import BookList from '../components/BookList';

const mockBooks = [
  {
    id: '001',
    title: 'Java',
    authors: ['Tom'],
    publisher: 'Oracle',
    publishDate: '2015/10/13',
    description: 'Java Core',
    pageCount: 200,
    categories: ['Java'],
    imageLinks: {
      smallThumbnail: '',
      thumbnail: '',
    },
    language: 'eng',
  },
  {
    id: '002',
    title: 'ReactJS',
    authors: ['Henry'],
    publisher: 'Facebook',
    publishDate: '2018/01/25',
    description: 'ReactJS for Beginners',
    pageCount: 180,
    categories: ['JavaScript'],
    imageLinks: {
      smallThumbnail: '',
      thumbnail: '',
    },
    language: 'eng',
  },
];

Enzyme.configure({ adapter: new Adapter() });

describe(`The BookList component test`, () => {
  it(`Should snapshot match`, () => {
    const wrapper = shallow(<BookList books={mockBooks} />);
    expect(wrapper).toMatchSnapshot();
  });

  it(`Must have 2 child components`, () => {
    const wrapper = shallow(<BookList books={mockBooks} />);
    expect(wrapper.children.length).toBe(1);
  });
});
