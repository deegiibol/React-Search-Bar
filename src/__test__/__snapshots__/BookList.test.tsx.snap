// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The BookList component test Should snapshot match 1`] = `ShallowWrapper {}`;
